src/store/generator/useGeneratorStore.ts

```diff
@@ -48,6 +48,7 @@ export const useGeneratorStore = create<GeneratorStore>()(
       recording = []
     ) =>
       set((state) => {
+        state.selectedRuleId = null
```

Comment:
It's not a big issue, because it only occurs when you have rules with the same id (so pretty much only when you have copies of the same generator file).

Reply:
We should probably use router for this, but it would be a bigger change